### PR TITLE
[pytorch][codev] add current test finding logic to find_matching_merge_rule

### PR DESCRIPTION
**Pull request by PaliC**
Summary:
Currently we enable skipping pytorch ci tests from codev which is analagous to "pytorchbot merge -f". I think it'd be reasonable to add functionality analagous to "pytorchbot merge -i" where we only skip currently failing tests. The easiest way to do this is to just have a boolean flag to trigger in find_matching_merge_rule and have the logic to find the currently failing tests there.

#FACEBOOK
D47485107 is the diff that adds a pytorch ci skipping label.

Test Plan: testing for trymerge should cover this change as its just a refactor.

Differential Revision: D47530806

### CODE REVIEW

File: .github/scripts/trymerge.py
Comment (ZainRizvi) [inline .github/scripts/trymerge.py:1262]: Oh man, this raw tuple hurts readability so much 💀

I know it's not your fault, but mind converting it to a NamedTuple?

**Review by ZainRizvi — APPROVED** [review]
LGTM, pending lint fixes and one minor nit.

File: .github/scripts/trymerge.py
Comment (huydhn) [inline .github/scripts/trymerge.py:1083]: As the flag is called `ignore_current`, let also use the same name here instead of `skip_current_checks` for consistency

File: .github/scripts/trymerge.py
Comment (huydhn) [inline .github/scripts/trymerge.py:1892]: Please double check if this works as expected as I have a gut feeling that having the below snipping running in `find_matching_merge_rule` would make the wrong behavior as this function is run inside a while loop:

```
checks = pr.get_checkrun_conclusions()
...
if skip_current_checks:
    _, failing = categorize_checks(checks, list(checks.keys()))
    ignore_current_checks = [x[0] for x in failing]
```

Basically, the while loop keeps running, so the list of failing checks could be growing.  Thus, `ignore_current_checks` could also grow.  This seems wrong as `merge -i` means ignoring only those failed checks at the time when the merge starts. cc @clee2000 for a second opinion

**Comment by clee2000** [comment]
I think this PR is old/not intended to be merged anymore since Sahan and I realized adding merge -i to codev doesn't make sense because we don't think codev/gh export checks waits like trymerge does. @PaliC should this be closed?

**Pull request closed without merging**
